transfers/transfers.go

```diff
@@ -279,6 +288,28 @@ func registerDatabases(conf config.Config) error {
 	return nil
 }
 
+func registerDatabaseByEndpointProvider(dbName string, dbConf map[string]any) error {
+	if epName, found := dbConf["endpoint"]; found {
+		epNameString, ok := dbConf["endpoint"].(string)
+		if !ok {
+			return fmt.Errorf("endpoint for database '%s' is not a string", dbName)
+		}
+		if ep, err := endpoints.NewEndpoint(epNameString); err == nil {
+			switch strings.Split(ep.Provider(), ":")[0] {
```

Comment:
I still don't see a need to carry additional information in the string returned by `Provider()`.